src/lib/mdns/minimal/ResponseSender.cpp

```diff
@@ -0,0 +1,159 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "ResponseSender.h"
+
+#include "QueryReplyFilter.h"
+
+#include <support/ReturnMacros.h>
+
+#define RETURN_IF_ERROR(err)                                                                                                       \
+    do                                                                                                                             \
+    {                                                                                                                              \
+        if (err)                                                                                                                   \
+        {                                                                                                                          \
+            return;                                                                                                                \
+        }                                                                                                                          \
+    } while (false)
+
+namespace mdns {
+namespace Minimal {
+
+namespace {
+
+constexpr uint16_t kMdnsStandardPort = 5353;
+constexpr uint16_t kPacketSizeBytes  = 512;
+
+} // namespace
+
+CHIP_ERROR ResponseSender::Respond(const QueryData & query, const chip::Inet::IPPacketInfo * querySource)
+{
+    mSendError = CHIP_NO_ERROR;
+
+    mCurrentSource = querySource;
+    mSendUnicast   = query.GetUnicastAnswer();
```

Comment:
I found Is reading oddly in this case, since this would be more like 'should the answer be unicast' rather than 'is this query a unicast answer'.

Renamed to RequestedUnicastAnswer. 'Is' prefix sounds odd to me in this context (even though I agree with it in the general case)